kubernetes/linux/setup.sh

```diff
@@ -44,8 +44,8 @@ sudo apt-get update
 sudo apt-get install td-agent-bit=1.6.8 -y
 
 # install ruby2.6
-sudo apt-get install software-properties-common -y
-sudo apt-add-repository ppa:brightbox/ruby-ng -y
+sudo apt-key adv --keyserver keyserver.ubuntu.com --recv-keys F5DA5F09C3173AA6
+sudo echo "deb http://ppa.launchpad.net/brightbox/ruby-ng/ubuntu bionic main" >> /etc/apt/sources.list
 sudo apt-get update
 sudo apt-get install ruby2.6 ruby2.6-dev gcc make -y
```

Comment:
on related note: why do we need gcc, make, ruby-dev (which also puts ruby docs) packages?

Reply:
> on related note: why do we need gcc, make, ruby-dev (which also puts ruby docs) packages?
these required to build the ruby c extensions and without these, we cant install fluentd-gem and these are minimal dependencies we would need to install for the fluentd-gem.  did very quick test, but didnt see ruby docs installed with ruby-dev.

Reply:
hmmm. i dont see any gcc or make dependencies (ruby depends on rake and not make).
Also, i dont fully understand why we need to install ruby this way.

'apt install ruby'  --> this just installs latest ruby runtime, and doesn't install ruby-docs, ruby-dev, gcc, make etc... i can install gems as well.

Reply:
in 18.04, "apt install ruby" install 2.5 which is included in the default ubuntu repo.  and ruby 2.6 is not part of the native ubuntu repo. Are you saying, you can install **ruby 2.6 wit apt install on 1804**? when I brought ruby 2.6 & fluentd gem updates, i picked the critical dependencies which are required but I can double check and see if we can get rid of any unneeded dependencies.

Reply:
double checked, we would need  all these dependencies ruby2.6-dev gcc make. gem part of the ruby2.6-dev and make and gcc required to build native extension.   I have image without ruby & fluentd - cidev:norubyandfluentd in case you want try out.
Error I was referring this
**Building native extensions. This could take a while...
ERROR:  Error installing fluentd:
        ERROR: Failed to build gem native extension.**

    current directory: /var/lib/gems/2.6.0/gems/msgpack-1.4.2/ext/msgpack
/usr/bin/ruby2.6 -I /usr/lib/ruby/2.6.0 -r ./siteconf20210805-901-10soqqt.rb extconf.rb
mkmf.rb can't find header files for ruby at /usr/lib/ruby/include/ruby.h

Reply:
so these are more fluentd dependencies than ruby  then ? also all these are install dependencies, After installing everything, we can remove all these dev tools, as we dont need them for runtime.

Reply:
> so these are more fluentd dependencies than ruby then ? also all these are install dependencies, After installing everything, we can remove all these dev tools, as we dont need them for runtime.

> yes, as I called out in my initial response and these are dependencies to install fluentd gem.  will have to play removing all these dependencies  after installing fluentd gem and see fluentd gem has any runtime dependencies coming from these dpendencies.

Reply:
i dont think there are any runtime dependencies. what you are doing is 'ruby-full'. what we need at runtime is 'ruby'

Reply:
looks like fluentd doesnt these dependencies for runtime so removed these build dependencies.

Reply:
Also, Ganga, when we remove packages, follow that with apt autoremove (as remove will only remove the packages asked to be removed and not its dependencies). So when you go apt autoremove does it remove more in this case ?